lib/component.js

```diff
@@ -616,12 +623,21 @@ class Component extends WebComponent {
       const attrType = attrSchema.type;
       let attrValue = null;
 
+      if (attrSchema.deprecatedMsg) {
+        console.warn(`${this}: attr '${attr}' is deprecated. ${attrSchema.deprecatedMsg}`);
+      }
+
       if (!this.hasAttribute(attr)) {
         if (attrType === `boolean` && attrSchema.default) {
           console.warn(
```

Comment:
I think this should throw a real error too. We can coalesce into single check


```ts
if (attrType === `boolean` && (attrSchema.default || attrsSchema.required)) {
  throw new Error(`${this}: boolean attr '${attr}' cannot have required or default, since its value is derived from whether dom element has the attribute, not its value`)
}
```